graphql/course/mutations.ts

```diff
@@ -106,6 +108,27 @@ export class MutateCourseResolver {
         }
         const result = await prisma.course.update({ data, where: { id: courseId } });
         logger.info(`Course (${result.id}) updated by Student (${context.user.studentId})`);
+
+        const subcourse = await prisma.subcourse.findFirst({ where: { courseId: courseId } });
+        const instructors = await prisma.subcourse_instructors_student.findMany({ where: { subcourseId: subcourse.id }, select: { student: true } });
+        const { context: contextData } = await prisma.user_achievement.findFirst({
+            where: { group: 'student_offer_course', context: { path: ['relation'], equals: `subcourse/${subcourse.id}` } },
+        });
+        contextData['courseName'] = result.name;
+        await prisma.user_achievement.updateMany({
```

Comment:
Are we sure that all the user achievements will have the same context? Maybe we should iterate of the list and update each individually.

Reply:
yes we are sure, because we selected the achievements of the same group. Achievements of the same group always share the same context.

Reply:
> yes we are sure, because we selected the achievements of the same group. Achievements of the same group always share the same context.

🤔 is this actually true? I thought that the achievement will get the context of the event, which might be different. I think I'm missing something. Can you please elaborate more on where/how this is happening?

Reply:
I did update this part, because I think you have a point here and I then was not satisfied the way I initially built it.
I'd appreciate you having another look at it.